test_requests.py

```diff
@@ -1177,6 +1181,24 @@ def test_precedence(self):
         encodings = requests.utils.get_encodings_from_content(content)
         assert encodings == ['HTML5', 'HTML4', 'XML']
 
+    def test_chunked_upload(self):
+        """ can safely send generators """
+        block_server = threading.Event()
+        server = Server.basic_response_server(wait_to_close_event=block_server)
+        data = (i for i in [b'a', b'b', b'c']) 
+
+        with server as (host, port):
+            url = 'http://{}:{}/'.format(host, port)
+            r = requests.post(url, data=data, stream=True)
+            block_server.set() # release server block
```

Comment:
Do we need this event? I think the server won't get closed until it has handled the request, so presumably we don't need to gate it.